actionview/lib/action_view/template/resolver.rb

```diff
@@ -10,8 +10,7 @@
 module ActionView
   # = Action View Resolver
   class Resolver
-    Path = ActionView::TemplatePath
-    deprecate_constant :Path
+    Path = ActiveSupport::Deprecation::DeprecatedConstantProxy.new("ActionView::Resolver::Path", "ActionView::TemplatePath", ActionView.deprecator)
```

Comment:
This converts a reference-time deprecation into a use-time one. Is that something we need here? I think we try to only use these as a last resort.

Reply:
> I think we try to only use these as a last resort.

I was not aware of that, but it makes sense.

In the case of `ActionView::Resolver::Path`, I think most occurrences of a reference would be accompanied by a use (e.g. `ActionView::Resolver::Path.build(...)`), but `DeprecatedConstantProxy` does break something like `class Foo < ActionView::Resolver::Path`.

I've changed this to use `DeprecatedConstantAccessor`'s `deprecate_constant` in #46580.